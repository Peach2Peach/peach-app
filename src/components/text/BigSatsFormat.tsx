import { ReactElement } from 'react'
import { TextStyle, View } from 'react-native'
import { Text } from '.'
import { SATSINBTC } from '../../constants'
import tw from '../../styles/tailwind'
import i18n from '../../utils/i18n'
import { getNumberFormatParts } from '../../utils/string'
import PaymentLogo from '../payment/PaymentLogo'

type SatsFormat = ComponentProps & {
  sats: number
  color?: TextStyle
}

export const BigSatsFormat = ({ sats, color, style }: SatsFormat): ReactElement => {
  const parts = getNumberFormatParts(sats / SATSINBTC)
  return (
    <View style={tw`flex flex-row`}>
      <PaymentLogo id="bitcoin" style={tw`mr-1 w-7 h-7`} />
      <Text style={[tw`h4`, parts[0] === '0' ? tw`text-black-5` : tw`text-black-1`]}>
        {parts[0]}.{parts[1]}
      </Text>
      <Text style={[tw`h4`, color || tw`text-black-1`, style]}>{parts[2]}</Text>
      <Text style={[tw`h6 ml-1 mt-0.5`, color || tw`text-black-1`, style]}>{i18n('currency.SATS')}</Text>
    </View>
  )
}

export default SatsFormat
