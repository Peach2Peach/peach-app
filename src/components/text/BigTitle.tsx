import React, { ReactElement } from 'react'
import { View } from 'react-native'
import { Text } from '.'
import tw from '../../styles/tailwind'
import Logo from '../../assets/logo/peachLogo.svg'

type BigTitle = {
  title: string
}

export const BigTitle = ({ title }: BigTitle): ReactElement => (
  <View style={tw`flex items-center`}>
    <Logo style={tw`w-40 h-40`} />
    <Text
      style={[
        tw`mt-3 text-2xl text-center uppercase font-baloo leading-3xl text-peach-1`,
        tw`mt-3 text-3xl text-center uppercase font-baloo leading-4xl text-peach-1`,
      ]}
    >
      {title}
    </Text>
  </View>
)

export default BigTitle
