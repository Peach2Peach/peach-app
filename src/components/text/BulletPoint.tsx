import { View } from 'react-native'
import tw from '../../styles/tailwind'
import { PeachText } from './Text'

type Props = { text: String }

export const BulletPoint = ({ text }: Props) => (
  <View style={tw`flex-row pl-3`}>
    <PeachText style={tw`text-xl body-m`}>Â· </PeachText>
    <PeachText style={tw`body-m`}>{text}</PeachText>
  </View>
)
