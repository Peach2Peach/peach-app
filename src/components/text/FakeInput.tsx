import React, { ReactElement } from 'react'
import {
  Pressable,
  View,
  ViewStyle
} from 'react-native'
import { TextInput } from 'react-native-gesture-handler'
import tw from '../../styles/tailwind'
import Icon from '../Icon'
import { Text } from '..'
import { Shadow } from 'react-native-shadow-2'
import { innerShadow } from '../../utils/layoutUtils'

interface FakeInputProps {
  value?: string,
  label?: string,
  icon?: string,
  isValid?: boolean,
  errorMessage?: string[],
  style?: ViewStyle|ViewStyle[]
}

/**
 * @description Component to display a fake input field
 * @param props Component properties
 * @param [props.value] current value
 * @param [props.label] input label
 * @param [props.icon] icon id
 * @param [props.isValid] if true show valid state
 * @param [props.errorMessage] error message for invalid field
 * @param [props.style] css style object
 * @example
 * <FakeInput
 *   value={address}
 *   label={i18n('form.btcAddress')}
 *   isValid={!isFieldInError('address')}
 *   errorMessage={getErrorsInField('address')}
 * />
 */
export const FakeInput = ({
  value,
  label,
  icon,
  isValid,
  errorMessage = [],
  style,
}: FakeInputProps): ReactElement => <View>
  <View style={tw`overflow-hidden rounded`}>
    <Shadow {...innerShadow} viewStyle={[
      tw`w-full flex flex-row items-center h-10 border border-grey-4 rounded pl-7 pr-3`,
      style ? style : {},
      isValid && value ? tw`border-green` : {},
      errorMessage.length > 0 ? tw`border-red` : {},
    ]}>
      <Text style={[tw`w-full flex-shrink p-0 text-grey-1 text-lg leading-5`]} numberOfLines={1} ellipsizeMode="middle">
        {value || label}
      </Text>
      {icon
        ? <Icon id="send" style={tw`w-5 h-5`} />
        : null
      }
    </Shadow>
  </View>

  {errorMessage.length > 0
    ? <Text style={tw`font-baloo text-xs text-red text-center mt-2`}>{errorMessage[0]}</Text>
    : null
  }
</View>

export default FakeInput