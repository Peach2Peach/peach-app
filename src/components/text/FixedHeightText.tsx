import { TextProps, View } from 'react-native'
import tw from '../../styles/tailwind'
import { PeachText } from './Text'

export const FixedHeightText = ({ height, style, ...textProps }: { height: number } & TextProps) => (
  <View style={{ height, justifyContent: 'center' }}>
    <PeachText style={[style, tw`-my-10`]} {...textProps} />
  </View>
)
