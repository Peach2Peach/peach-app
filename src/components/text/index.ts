export { BigTitle } from './BigTitle'
export { Headline } from './Headline'
export { PeachText as Text } from './Text'
export { SatsFormat } from './SatsFormat'
export { TextLink } from './TextLink'
export { Timer } from './Timer'
export { Title } from './Title'
