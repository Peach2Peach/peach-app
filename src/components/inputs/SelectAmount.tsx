import React, { ReactElement } from 'react'
import { View } from 'react-native'
import tw from '../../styles/tailwind'
import { innerShadow } from '../../utils/layout'
import { Text } from '../text'
import { Shadow } from '../ui'
import Input from './Input'

type SelectAmountProps = ComponentProps & {
  min: number
  max: number
  value: number
  onChange: (value: number) => void
}

/**
 * @description Component to display a range selection
 * @example
 * <SelectAmount
    min={100000}
    max={2500000}
    value={amount}
    onChange={setAmount}
 */
export const SelectAmount = ({ min, max, value, onChange }: SelectAmountProps): ReactElement => {
  const change = (amount: string) => onChange(Number(amount))

  return (
    <View>
      <Shadow shadow={innerShadow} style={tw`flex items-center p-4 bg-primary-background-dark`}>
        <View style={tw`flex-row`}>
          <Input
            {...{
              label: 'amount',
              placeholder: 'amount',
              value: value.toString(),
              onChange: change,
              keyboardType: 'numeric',
            }}
          />
        </View>
      </Shadow>
      <View style={tw`mt-10`}>
        <Text>Min: {min}</Text>
        <Text>max {max}</Text>
        <Text>Current {value}</Text>
      </View>
    </View>
  )
}
