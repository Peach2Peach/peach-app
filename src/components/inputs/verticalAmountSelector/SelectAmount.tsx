import { Animated, View } from 'react-native'
import tw from '../../../styles/tailwind'
import i18n from '../../../utils/i18n'
import { getTranslateY } from '../../../utils/layout'
import { ParsedPeachText } from '../../text'
import { CustomAmount } from './CustomAmount'
import { SliderKnob } from './SliderKnob'
import { SliderTrack } from './SliderTrack'
import { TrackMarkers } from './TrackMarkers'
import { knobLayout } from './helpers/knobLayout'
import { onStartShouldSetResponder } from './helpers/onStartShouldSetResponder'
import { useSelectAmountSetup } from './hooks/useSelectAmountSetup'

type Props = ComponentProps & {
  min: number
  max: number
  value: number
  onChange: (value: number) => void
}

export const SelectAmount = ({ min, max, value, onChange, children, style }: Props) => {
  const { amount, updateCustomAmount, pan, panResponder, trackRange, onTrackLayout } = useSelectAmountSetup({
    min,
    max,
    value,
    onChange,
  })
  return (
    <View style={[tw`flex-row items-center justify-between`, style]}>
      <View style={[tw`items-start gap-2 shrink`, tw`md:gap-4`]}>
        <ParsedPeachText
          style={[tw`h7`, tw`md:h5`]}
          parse={[{ pattern: new RegExp(i18n('sell.subtitle.highlight'), 'u'), style: tw`text-primary-main` }]}
        >
          {i18n('sell.subtitle')}
        </ParsedPeachText>
        <CustomAmount amount={amount} onChange={updateCustomAmount} style={tw`items-start flex-shrink`} />
        {children}
      </View>
      <SliderTrack style={tw`h-full`} onLayout={onTrackLayout}>
        <TrackMarkers />
        <Animated.View
          {...panResponder.panHandlers}
          {...{ onStartShouldSetResponder }}
          style={[knobLayout, getTranslateY(pan, trackRange)]}
        >
          <SliderKnob />
        </Animated.View>
      </SliderTrack>
    </View>
  )
}
