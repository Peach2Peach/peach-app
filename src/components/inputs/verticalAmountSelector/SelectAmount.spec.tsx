import { createRenderer } from 'react-test-renderer/shallow'
import { SelectAmount } from './SelectAmount'
import { Text } from '../../text'

describe('SelectAmount', () => {
  const shallowRenderer = createRenderer()
  it('should render correctly', () => {
    shallowRenderer.render(<SelectAmount min={50000} max={5000000} value={60000} onChange={jest.fn()} />)
    expect(shallowRenderer.getRenderOutput()).toMatchSnapshot()
  })
  it('should render children correctly', () => {
    shallowRenderer.render(
      <SelectAmount min={50000} max={5000000} value={60000} onChange={jest.fn()}>
        <Text>I'm a child</Text>
      </SelectAmount>,
    )
    expect(shallowRenderer.getRenderOutput()).toMatchSnapshot()
  })
})
