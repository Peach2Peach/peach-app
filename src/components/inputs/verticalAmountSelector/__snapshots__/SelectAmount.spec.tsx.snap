// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectAmount should render correctly 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingLeft": 20,
        "paddingRight": 16,
      },
      undefined,
    ]
  }
>
  <View
    style={
      [
        {
          "alignItems": "flex-start",
          "flexShrink": 1,
          "gap": 8,
        },
        {
          "gap": 16,
        },
      ]
    }
  >
    <ParsedPeachText
      parse={
        [
          {
            "pattern": /selling/u,
            "style": {
              "color": "#F56522",
            },
          },
        ]
      }
      style={
        [
          {
            "fontFamily": "Baloo2-Bold",
            "fontSize": 16,
            "letterSpacing": 0.16,
            "lineHeight": 26,
          },
          {
            "fontFamily": "Baloo2-Bold",
            "fontSize": 24,
            "letterSpacing": 0.24,
            "lineHeight": 32,
          },
        ]
      }
    >
      You're selling:
    </ParsedPeachText>
    <CustomAmount
      amount={60000}
      onChange={[Function]}
      style={
        {
          "alignItems": "flex-start",
          "flexShrink": 1,
        }
      }
    />
  </View>
  <SliderTrack
    onLayout={[Function]}
    style={
      {
        "height": "100%",
      }
    }
  >
    <TrackMarkers />
    <ForwardRef(AnimatedComponentWrapper)
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        [
          {
            "alignItems": "center",
            "flexDirection": "row",
            "marginLeft": -24,
            "marginRight": -24,
            "paddingLeft": 24,
            "paddingRight": 24,
            "position": "absolute",
            "top": 0,
          },
          {
            "transform": [
              {
                "translateY": 230,
              },
            ],
          },
        ]
      }
    >
      <SliderKnob />
    </ForwardRef(AnimatedComponentWrapper)>
  </SliderTrack>
</View>
`;
